kernel/src/process.rs

```diff
@@ -400,6 +435,26 @@ pub trait Process {
     /// The offset indicates the multiple of 64 command numbers to get permissions for.
     fn get_command_permissions(&self, driver_num: usize, offset: usize) -> CommandPermissions;
 
+    /// Get the process `write_id`.
+    /// Returns `None` if a `write_id` is not included.
+    fn get_write_id(&self) -> Option<u32>;
+
+    /// Get the `read_ids`.
+    /// Returns `None` if a `read_ids` is not included.
+    fn get_read_ids(&self) -> Option<[u32; 8]>;
+
+    /// Get the number of `read_ids`.
+    /// Returns `None` if a `read_ids` is not included.
+    fn num_read_ids(&self) -> Option<usize>;
+
+    /// Get the `access_ids`.
+    /// Returns `None` if a `access_ids` is not included.
+    fn get_access_ids(&self) -> Option<[u32; 8]>;
+
+    /// Get the number of `access_ids`.
+    /// Returns `None` if a `access_ids` is not included.
+    fn num_access_ids(&self) -> Option<usize>;
+
```

Comment:
All boards pay for the code size of this support, even if they do not use the KV system. Looks like it is about 250 bytes per board, which is of course not a ton. I think long-term it is important for Tock to find ways to expose TBF header extensions to capsules in a configurable manner, but won't ask to block this PR on that support

Reply:
Hopefully long term we can eliminate virtual functions with https://github.com/rust-lang/rust/issues/68262